DeepLabCut/deeplabcut_main.py

```diff
@@ -0,0 +1,159 @@
+
+    #--------------- Save extracted images (used for labeling bodyparts) as .png ---------------
+    cap = cv2.VideoCapture(video_path)
+    frame_count = 0
+    while cap.isOpened():
+        ret, frame = cap.read()
+        if not ret:
+            break 
+        if frame_count in [int(item) for item in extracted_frames]:
+            # Save the frame as a PNG file
+            frame_filename = f'{labeled_datapath}/img{frame_count}.png'
+            cv2.imwrite(frame_filename, frame)
+        frame_count += 1 
+    cap.release() 
+    cv2.destroyAllWindows()
+
+def update_config_file(path_config_file, bodypart_names):
+    # Load the YAML file
+    with open(path_config_file, 'r') as file:
+        data = yaml.safe_load(file)
+
+    # Modify the specific text   
+    data['bodyparts'] = bodypart_names
+
+    # Save the modified YAML back to the file
+    with open(path_config_file, 'w') as file:
+        yaml.safe_dump(data, file, default_flow_style=False)
+
+def save_coords_to_doric(filepath, datapath, path_output, deeplabcut_params, group_names):
+    file_           = h5py.File(filepath, 'a')
+    bodypart_names  = deeplabcut_params.params.pop(dlc_defs.Parameters.danse.BODY_PART_NAMES).split(', ')
+    bodypart_colors = deeplabcut_params.params.pop(dlc_defs.Parameters.danse.BODY_PART_COLORS).split(', ')
+    deeplabcut_params.params[dlc_defs.Parameters.danse.VIDEO_DATAPATH] = datapath
+
+    data, group, operation, series, videoName = group_names
+    time_ = np.array(file_[f"{datapath}/{defs.DoricFile.Dataset.TIME}"])
+    h5_files = glob.glob(os.path.join(path_output, '*.h5'))
+    file_path = h5_files[0]
+    data_coords = pd.read_hdf(file_path)
+    operation_path = f'{group}/{dlc_defs.Parameters.danse.COORDINATES}/{series}/{videoName}{dlc_defs.DoricFile.Group.POSE_ESTIMATION}'
+
+    for index, bodypart in enumerate(bodypart_names):
+        coords = data_coords.loc[:, pd.IndexSlice[:, bodypart, ['x','y']]]
+        coords_df = pd.DataFrame(coords.values)
+        dataset_path = f'{operation_path}/{bodypart}'
+
+        if dataset_path in file_: 
+            del file_[dataset_path] # Remove existing dataset if it exists 
+        file_.create_dataset(dataset_path, data=coords_df, dtype = 'int32', chunks=utils.def_chunk_size(coords_df.shape), maxshape=(h5py.UNLIMITED, 2))
+        attrs = {
+            defs.DoricFile.Attribute.Dataset.USERNAME: bodypart,
+            defs.DoricFile.Attribute.ROI.COLOR       : bodypart_colors[index]
+        }
+
+        time_path = f'{group}/{dlc_defs.Parameters.danse.COORDINATES}/{series}/{videoName}{dlc_defs.DoricFile.Group.POSE_ESTIMATION}/{defs.DoricFile.Dataset.TIME}'
+        if time_path not in file_:
+            file_.create_dataset(time_path, data=time_)
+
+        utils.save_attributes(attrs, file_, dataset_path)
+
+    utils.save_attributes(utils.merge_params(params_current = deeplabcut_params.params), file_, operation_path)
+    file_.close()
```

Comment:
Missing the print at the end of this function, please review saving functions from the other python packages